### ISSUE TITLE
Can't run `./gradlew mappings` on java 19

### ISSUE THREAD

**Issue by TheGlitch76** [QuiltMC/quilt-mappings#266]
Here is the output of `./gradlew mappings` on my machine:
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not open cp_settings generic class cache for settings file '/home/glitch/Projects/Mappings/quilt-mappings/settings.gradle' (/home/glitch/.gradle/caches/7.5.1/scripts/1isd85hnmm825pv61wywqj19v).
> BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 63

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 443ms
```

I assume this is because I am running Java 19, and QM's buildscript is stuck on Gradle 7.

**Comment by triphora** [comment]
This isn't a QM issue. The stable version of Gradle itself doesn't support Java 19.

**Comment by EnnuiL** [comment]
Oh, this is only the first step on the labyrinth of Lack of Java 17 Support: the Gradle Gate; Updating to Gradle 7.6-rc1 shows that QM does have some updating to do, specifically? Updating the ASM dep to 9.4 on Enigma, Unpick, and possibly the others;
Yes, we can't update to a Gradle RC yet, but there's still other things to do

**Comment by EnnuiL** [comment]
Gradle 7.6 has been released; I've gone ahead and brought [Unpick (#2)](https://github.com/QuiltMC/unpick/pull/2) and [Enigma (#18)](https://github.com/QuiltMC/enigma/pull/18) up to speed; Those should be the only blockers for Java 19 compatibility
